### ISSUE TITLE
Navigating to "commits" tab should have visual feedback

### ISSUE THREAD

**Issue by rudolfs** [radicle-dev/radicle-upstream#920]
On larger projects with many commits there is a substantial delay between clicking on the "commits" tab and actually navigating there. We should at least indicate that there is some activity to the user via a waiting cursor.

Refs: https://github.com/radicle-dev/radicle-upstream/issues/912.

![commits](https://user-images.githubusercontent.com/158411/93457541-6854ba00-f8df-11ea-8f94-bd9a63fb2bb0.gif)

**Comment by sarahscott** [comment]
@juliendonck @brandonhaslegs what are you thinking for this one?

**Comment by brandonhaslegs** [comment]
We've just change the cursor to a `cursor: wait;` in other scenarios like this.

**Comment by MeBrei** [comment]
With the new implementation, the commit tab is moved into the horizontal menu which uses the `svelte-spa-router` and the commits are fetched to show the correct commit count. Before, the commit/branches/contributors tab had a custom routing which allowed to add a click handler to delay switching to the commit tab until it was fully loaded. 
With the new implementation, I can see the following options for this:

1. showing `cursor: wait` after clicking the commit tab. It would already navigate to the commit page and show an empty screen (besides the header etc) until the commits are loaded
2. showing `cursor: wait` on hover and/or disabling the commit tab until the commits are loaded.

**Comment by rudolfs** [comment]
I like option nr. 2.

**Comment by rudolfs** [comment]
To clarify, by "showing cursor: wait on *hover* and/or disabling", do you mean:
1. user hovers the commits tab, cursor switches to *pointer*
2. user clicks the commits tab, cursor switches to *wait*, but we stay on the same screen until the commits are loaded from the backend
3. we display the commits and return the cursor to *auto*

**Comment by MeBrei** [comment]
> To clarify, by "showing cursor: wait on _hover_ and/or disabling", do you mean:
> 
> 1. user hovers the commits tab, cursor switches to _pointer_
> 2. user clicks the commits tab, cursor switches to _wait_, but we stay on the same screen until the commits are loaded from the backend
> 3. we display the commits and return the cursor to _auto_

I meant that the cursor shows wait or disabled on hover and the tab is not clickable. Once the commits are successfully loaded, the commit tab can be clicked and the cursor is on auto. 
I haven't found a way yet to delay the switching of screens with the `svelte-spa-router` and adding a custom click handler to somehow prevent the router from switching immediately seems odd to me.

**Comment by rudolfs** [comment]
Ooooh, you mean you would load the commits already when landing on the project source page, and disable the tab until they're loaded?

Hmm, in that case I think we would some more visual feedback than just a spinning cursor on hover. If the tab is disabled and greyed out initially, I'd be surprised that it suddenly turns enabled.

**Comment by MeBrei** [comment]
Yes, the commits are already fetched via the main project page, as it also returns the number of commits that is shown in the tab. The horizontal menu shows `Commit` until the request was successful, then it shows `Commit <count>`.

**Comment by rudolfs** [comment]
I wonder if we maybe could further improve the UX by:
- making the tab disabled (greyed out and cursor *wait*)
- showing the commit count badge, but without the count and with a [content placeholder](https://www.codepicky.com/content-placeholders/) shimmer

![placeholder-loading-demo-1](https://user-images.githubusercontent.com/158411/94674940-3b5dc980-0319-11eb-9162-9b76d4bcefa0.gif)

I think I'm still fine with option 2 and @brandonhaslegs should have the last call on any micro interactions.

**Comment by brandonhaslegs** [comment]
#1 Is more of a normal solution right? Click nav link, wait for page to load.

#2 Prevents you from clicking the Commit nav item until the commits are loaded? Sounds strange, but honestly, a user will not even notice this unless they click on commits immediately after arriving on the project page. But if they do, I think it would be confusing to see a ⏳ when hovering over a nav link. It's not a normal thing. to prevent navigation until the page you're going to has loaded.

I think we should do #1. And we should definitely do content placeholders, but I guess that's not a Beta 1 feature.

**Comment by rudolfs** [comment]
To add a bit more context, this is what it looks like at the moment, when working with a small project:
![slow2](https://user-images.githubusercontent.com/158411/94676529-c3dd6980-031b-11eb-9dac-7fec3888958a.gif)

This is what happens if I try to immediately navigate to the tab before comments could be loaded:
![slow1](https://user-images.githubusercontent.com/158411/94676565-d5267600-031b-11eb-971c-36dbf91fadac.gif)

And this happens on a bigger project:
![slooooow](https://user-images.githubusercontent.com/158411/94676726-10c14000-031c-11eb-8de2-24a1ff59e6e9.gif)

We definitely need to look into our performance: https://github.com/radicle-dev/radicle-upstream/issues/512.

**Comment by brandonhaslegs** [comment]
Why don't commits load if you have the project replicated? Why is there any loading at all?

**Comment by rudolfs** [comment]
@brandonhaslegs because we haven't had the chance to optimise the code: #512.

**Comment by MeBrei** [comment]
Summary of outside conversation:
* for beta-1, keep the navigation behavior as is, but add `cursor: wait` while the commits are loading (low priority)
* after beta-1 navigation & loading states will be revised more generally

**Comment by xla** [comment]
Resolved in #1080
